pkg/vsphere/tasks/waiter.go

```diff
@@ -91,15 +91,19 @@ func WaitForResult(ctx context.Context, f func(context.Context) (Task, error)) (
 
 func isTaskInProgress(err error) bool {
 	if soap.IsSoapFault(err) {
-		if _, ok := soap.ToSoapFault(err).VimFault().(types.TaskInProgress); ok {
+		f := soap.ToSoapFault(err)
+		if _, ok := f.VimFault().(types.TaskInProgress); ok {
 			return true
 		}
+		log.Debugf("unexpected soap fault on task retry : %s", spew.Sdump(f))
```

Comment:
Just to clarify - we do want the improved logging that comes from unwrapping the error from Soap/Vim as you've done here.  The type returned by `f.VimFault()` here, `soap.ToVimFault()` and `err.Fault()` below are at the level we want.

Versus the line you removed `log.Errorf("task failed: %s", err)` where we weren't getting any detail on the error type.